### PR TITLE
Add scalar autograd tests for functions requiring 'special' Variables…

### PR DESCRIPTION
**Pull request by gchanan**
… on LHS.

### CODE REVIEW

File: torch/autograd/gradcheck.py [hunk 1]
```diff
@@ -203,6 +203,10 @@ def fn(input):
                 continue
             if not i.grad.data.eq(0).all():
                 return fail_test('backward not multiplied by grad_output')
+            if type(i.grad) != type(i):
```
Comment (colesbury) [inline torch/autograd/gradcheck.py]: Use `i.grad.type()` and `i.type()` so that it will work with variables

**Review by colesbury — APPROVED** [review]

**Comment by colesbury** [comment]
I don't think we should allow scalars in `addr` unless we're going to add support for N-dimensional tensors.

**Comment by gchanan** [comment]
That same logic should apply to `ger` and should be easy to do assuming we are willing to get rid of the `out` variants -- do you want to do that?

**Comment by gchanan** [comment]
ok, I'm going to back out the contentious stuff; anything we do would probably be done as a native function, where we would want `out` support; I'll add that in a separate PR and we'll decide what to do after that.

**Comment by gchanan** [comment]
Onnx failure looks unrelated.

**Pull request merged**